scheduler/reconcile.go

```diff
@@ -224,12 +255,13 @@ func (a *allocReconciler) computeGroup(group string, all allocSet) {
 		a.markStop(migrate, "", allocNotNeeded)
 		a.markStop(lost, structs.AllocClientStatusLost, allocLost)
 		desiredChanges.Stop = uint64(len(untainted) + len(migrate) + len(lost))
-		return
+		return true
 	}
 
 	// Get the deployment state for the group
 	var dstate *structs.DeploymentState
-	var existingDeployment bool
+	existingDeployment := false
+	deploymentComplete := true
```

Comment:
Why not default to false and then just check for the case where we make any allocations to set to true? Seems like way more cases are false than true